src/internal/ans.ts

```diff
@@ -0,0 +1,192 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * This file contains the underlying implementations for exposed API surface in
+ * the {@link api/name}. By moving the methods out into a separate file,
+ * other namespaces and processes can access these methods without depending on the entire
+ * name namespace and without having a dependency cycle error.
+ */
+
+import { AptosConfig } from "../api/aptosConfig";
+import { MoveOption, MoveString, U64 } from "../bcs";
+import { Account, AccountAddress } from "../core";
+import { InputGenerateTransactionOptions, InputSingleSignerTransaction } from "../transactions/types";
+import { HexInput, MoveAddressType, MoveValue } from "../types";
+import { Network } from "../utils/apiEndpoints";
+import { view } from "./general";
+import { generateTransaction } from "./transactionSubmission";
+
+export type ANSName = string | { domainName: string; subdomainName?: string };
```

Comment:
why do accept object as an ans name input?

Reply:
It was my initial implementation, then we shifted to a more concise `string` variant. I thought to keep it around in case the developer has a preference of one way or the other, but can consolidate into just the string if you prefer.

Reply:
Removed the type and are just using strings now!